src/Sentry/Protocol/Span.cs

```diff
@@ -0,0 +1,105 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using Sentry.Internal.Extensions;
+
+namespace Sentry.Protocol
+{
+    public class Span : ISpan, IJsonSerializable
+    {
+        public SentryId SpanId { get; set; }
+        public SentryId? ParentSpanId { get; set; }
+        public SentryId TraceId { get; set; }
+        public DateTimeOffset StartTimestamp { get; set; }
+        public DateTimeOffset EndTimestamp { get; set; }
+        public string? Operation { get; set; }
+        public string? Description { get; set; }
+        public string? Status { get; set; }
+        public bool IsSampled { get; set; }
+
+        private ConcurrentDictionary<string, string>? _tags;
+        public IReadOnlyDictionary<string, string> Tags => _tags ??= new ConcurrentDictionary<string, string>();
```

Comment:
I assume concurrency here might be more of an issue so this approach of instantiating on the getter won't work well here.

We might need to synchronize in order to make sure concurrent calls to `get` get the same instance. Maybe a double check lock will do

Reply:
So Spans have tags? I was thinking that only Transactions could have tags

Reply:
Maybe just eagerly initialize the instance? It will probably be a smaller overhead than a lock.

Reply:
one will incur overhead on each new span due to allocating the list then GC'ing it. The other one will cost when accessing it only and only the first time due to the double check lock.

Without measuring we're just guessing, the the transaction I'm happy with always having the list there since more often than not we'll use it, for individual Spans, I don't know

Reply:
> The other one will cost when accessing it only and only the first time due to the double check lock.

Maybe I misunderstood your suggestions, but how will that work? From my understand, the lock has to be on the getter, which would mean it's acquired/checked every time the property is accessed.

Reply:
https://en.wikipedia.org/wiki/Double-checked_locking

Reply:
@lucas-zimerman the docs say so, but I also found the docs to be incredibly unreliable too, so...
https://develop.sentry.dev/sdk/event-payloads/span/

Reply:
If the docs are not good, we should fix it. Please open a PR there or at least raise a flag (open an issue asking for clarification).